dbt-bigquery/src/dbt/adapters/bigquery/relation.py

```diff
@@ -95,7 +95,33 @@ def materialized_view_config_changeset(
         config_change_collection = BigQueryMaterializedViewConfigChangeset()
         new_materialized_view = cls.materialized_view_from_relation_config(relation_config)
 
-        if new_materialized_view.options != existing_materialized_view.options:
+        # Compare options safely by converting them to comparable format
+        def _options_equal(options1, options2):
+            """Compare two BigQueryOptionsConfig objects safely."""
+            if options1 is None and options2 is None:
+                return True
+            if options1 is None or options2 is None:
+                return False
+
+            # Get all possible field names from both objects
+            all_fields = set(options1.__dict__.keys()) | set(options2.__dict__.keys())
+
+            # Compare all fields
+            for key in all_fields:
+                if key in ("labels", "tags"):
+                    # Compare dict fields separately, treating None as distinct from {}
+                    val1 = getattr(options1, key, None)
+                    val2 = getattr(options2, key, None)
+                    if val1 != val2:
+                        return False
+                else:
+                    val1 = getattr(options1, key, None)
+                    val2 = getattr(options2, key, None)
+                    if val1 != val2:
+                        return False
+            return True
```

Comment:
pretty sure this can be simplified down using some iterators:
```suggestion
     if a is b:                     # covers (a, b) is (None, None)
        return True
    if a is None or b is None:     # exactly one is None
        return False

    # Union of attribute names present on either side
    keys = a.__dict__.keys() | b.__dict__.keys()

    return all(getattr(a, k, None) == getattr(b, k, None) for k in keys)
```